pulpcore/pulpcore/app/viewsets/content.py

```diff
@@ -61,6 +61,23 @@ def create(self, request):
         headers = self.get_success_headers(serializer.data)
         return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
 
+    def destroy(self, request, pk):
+        """
+        Remove Content unit only if it is not associated with a repository version
+        """
+        repository_content_set = RepositoryContent.objects.filter(content__pk=pk)
+        
```

Comment:
I'm wondering whether it's OK for repository content, that was removed in particular version, should be prevented from being destroyed. In case it shouldn't, the filter can be expressed like this:
```python
query = models.Q(content__pk=pk) & (models.Q(version_removed=None) |
                                    models.Q(version_removed__lt=models.F('version_added')))
if RepositoryContent.objects.filter(query):
    ...
```
Ideally, an explanation why this kind of filtering isn't necessary would be provided (as a comment in the code) for further reference.

Reply:
`version_removed` is never going to be less than `version_added`. If a content unit gets re-added to a repository, a new RepositoryContent record gets created.